docs/readmes/orc8r/dev_sub_digests.md

```diff
@@ -0,0 +1,88 @@
+
+### AGW Digest Store
+
+Aside from its subscriber database, a gateway also manages SQL stores for its local flat and per-subscriber digests.
+
+Every time a gateway engages in a sync session with the cloud, it receives the latest cloud digests, with which it updates its local SQL store. These digests represent the version of subscriber data stored in the gateway, and they are used in the gateway's subsequent requests to the cloud.
+
+### Cloud-AGW Callpath
+
+The current interactions between the `subscriberdb` cloud servicer and the AGW client involve 3 servicer endpoints
+
+1. `CheckSubscribersInSync`
+    - AGW sends its local flat digest
+    - Cloud compares the AGW digest with the cloud digest. Depending on whether they are the same, the cloud returns a signal for whether the AGW is in-sync
+
+2. If AGW is not in-sync: `SyncSubscribers`
+    - AGW sends its local per-subscriber digests
+    - Cloud compares the AGW digests with the cloud digests, and calculates the changeset between the two sets of per-subscriber digests. That is, it locates the subscribers to renew or to delete on the AGW side
+        - If the changeset is of a reasonable size (configurable), returns the data needed to update AGW locally, as well as the latest cloud digests
+        - If not, returns a signal for AGW to conduct a full-on resync
+3. If AGW receives resync signal: `ListSubscribers`
+    - AGW sends desired pageSize, as well as the current page token (until all pages are fetched)
+    - Cloud returns the corresponding page of data loaded from `subscriberdb_cache`, as well as the latest cloud digests
+
+## Additional Notes
+
+### APN Resource Configs Handling
+
+Currently, gateway-specific APN resources configurations are also stored in subscriber data objects streamed down to AGWs. This is an issue to be addressed by ongoing refactoring projects.
+
+Until a fix is landed, to ensure that the generated subscriber digests are the same for all gateways in a network, the APN resources configs are extracted from subscriber data objects beforehand, and captured in its own digest instead. That is, the subscriber digests generated in `subscriberdb_cache` are concatenations of `apnResourceDigest` and `subscriberDigest`; in the generation of the `subscriberDigest`, the APN resource configurations are left blank.
+
+Similarly, the cached subscriber data objects in `subscriberdb_cache` also don't contain APN resource configs. As a result, the `subscriberdb` servicers would currently append APN resource configs to subscriber data objects loaded from cache, before streaming it down to AGWs.
+
+### Orc8r-directed Resync
+
+For reliability, and also to avoid hash collision-induced edge cases in the subscriber digests pattern, an orc8r-directed resync is enforced on the AGWs at a configurable interval (e.g. once per day).
```

Comment:
```suggestion
For added reliability, and to avoid (trivially improbable) hash collisions, an Orc8r-directed resync is enforced on the AGWs at a configurable interval (e.g. once per day).
```